internal/surfer/gcloud/generate.go

```diff
@@ -17,35 +17,154 @@ package gcloud
 import (
 	"context"
 	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
 
-	"github.com/googleapis/librarian/internal/sidekick/config"
-	"github.com/googleapis/librarian/internal/sidekick/parser"
+	"github.com/googleapis/librarian/internal/sidekick/api"
+	"github.com/googleapis/librarian/internal/surfer/gcloud/utils"
 	"github.com/googleapis/librarian/internal/yaml"
 )
 
 // Generate generates gcloud commands for a service.
-func Generate(ctx context.Context, googleapis, gcloudconfig, output string) error {
-	cfg, err := yaml.Read[Config](gcloudconfig)
+func Generate(_ context.Context, googleapis, gcloudconfig, output, includeList string) error {
+	overrides, err := readGcloudConfig(gcloudconfig)
 	if err != nil {
 		return err
 	}
 
-	model, err := parser.ParseProtobuf(&config.Config{
-		General: config.GeneralConfig{
-			// TODO(https://github.com/googleapis/librarian/issues/2817):
-			// determine the specification source
-			SpecificationSource: "",
-		},
-		Source: map[string]string{
-			"googleapis-root": googleapis,
-		},
-	})
+	model, err := createAPIModel(googleapis, includeList)
 	if err != nil {
-		return fmt.Errorf("failed to create API model: %w", err)
+		return err
+	}
+
+	if len(model.Services) == 0 {
+		return fmt.Errorf("no services found in the provided protos")
+	}
+
+	for _, service := range model.Services {
+		// TODO(https://github.com/googleapis/librarian/issues/3291): Ensure output directories don't collide if multiple services share a name.
+		if err := generateService(service, overrides, model, output); err != nil {
+			return fmt.Errorf("failed to generate commands for service %q: %w", service.Name, err)
+		}
+	}
+	return nil
+}
+
+func generateService(service *api.Service, overrides *Config, model *api.API, output string) error {
+	// Determine short service name for directory structure.
+	// The `shortServiceName` is derived from `service.DefaultHost` (e.g., "parallelstore.googleapis.com" -> "parallelstore").
+	// `service.DefaultHost`  matches the name field in the service config file
+	// (e.g., `default_host` for parallelstore is derived from `parallelstore_v1.yaml` name field).
+	shortServiceName := ""
+	hostParts := strings.Split(service.DefaultHost, ".")
+	if len(hostParts) > 0 {
```

Comment:
shortServiceName, found, _ := strings.Cut(service.DefaultHost, ".")
	if !found {
		return fmt.Errorf("failed to determine short service name for service %q: default_host is empty", service.Name)
	}